iocore/eventsystem/I_MIOBufferWriter.h

```diff
@@ -0,0 +1,174 @@
+#if !defined I_MIOBUFFERWRITER_H_
+#define I_MIOBUFFERWRITER_H_
+
+/** @file
+
+    Buffer Writer for an MIOBuffer.
+
+    @section license License
+
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+ */
+
+#include <cstring>
+
+#include <ts/ink_assert.h>
+#include <ts/BufferWriter.h>
+
+#if !defined(UNIT_TEST_BUFFER_WRITER)
```

Comment:
I'd rather have the unit text binary link with the event system library than have conditional includes just for testing in the implementation header file. If you don't want to do the link, then in the test file `#define` the `I_IOBuffer.h` header guard before you include this header.

Reply:
A lot of work for dubious benefit in my opinion but anyway.

Reply:
It's less code and work than doing it this way.

Reply:
Its certainly less code for you since I have to write it.  But its more code for me, more mocking.  Particularly since it's already done the other way.

Reply:
I don't see that. You remove these 3 lines and replace them with one in the test file,
```
#define I_IOBUFFER_h
```
That accomplishes the same thing as far as the test code is concerned.

Reply:
Oh OK, I thought you wanted me to use the actual contents of I_IOBuffer.h in the UT.

Reply:
If IOBuffer.h is changed to use #pragma once , what you're suggesting will break.

Reply:
Yes. That's a bit of a concern to me, that I think I will bring up at the summit. It works for now, though.